.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      name: Checkout code
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.12'
+
+    - name: Install uv
+      run: |
+        python -m pip install --upgrade pip
+        pip install uv
+
+    - name: Set up environment with uv
+      run: |
+        uv venv
+        uv pip install --no-cache-dir -r requirements-dev.lock
+
+    - name: Add .venv to PATH
+      run: echo "$GITHUB_WORKSPACE/.venv/bin" >> $GITHUB_PATH
```

Comment:
Why did we have to do this?

Reply:
Idk that's how I saw it in the other one so I assumed it was just the standard set up for python

But let me know.

Reply:
Ok, GITHUB actions require access to the `.venv`. 

Past me was smarter than present me